apps/test/unit/lib/kits/maker/portScanningTest.js

```diff
@@ -8,9 +7,43 @@ import {
   OSX_DEFAULT_PORTS,
   OTHER_BAD_SERIALPORTS
 } from './sampleSerialPorts';
-import BoardController from '@cdo/apps/lib/kits/maker/BoardController';
+import ChromeSerialPort from './FakeChromeSerialPort';
+import {
+  findPortWithViableDevice,
+  getPreferredPort
+} from '@cdo/apps/lib/kits/maker/portScanning';
+
+describe("Maker Toolkit", function () {
+  describe(`findPortWithViableDevice()`, () => {
+    // Testing against FakeChromeSerialPort.js
+    afterEach(() => {
+      ChromeSerialPort.test_reset();
+    });
+
+    it('resolves with a port if a viable device is found', done => {
+      ChromeSerialPort.test_setDeviceList(CIRCUIT_PLAYGROUND_PORTS);
+      findPortWithViableDevice()
+          .then(port => {
+            expect(port).to.equal('COM5');
+            done();
+          })
+          .catch(done);
+    });
+
+    it('rejects if no viable device is found', done => {
+      ChromeSerialPort.test_setDeviceList(OTHER_BAD_SERIALPORTS);
+      findPortWithViableDevice()
+          .then(port => {
+            done(new Error('Expected promise to reject, but it resolved to ' + port));
+          })
+          .catch(err => {
+            expect(err.message).to.equal('Did not find a usable device on a serial port.');
+            done();
+          })
+          .catch(done);
```

Comment:
Maybe non-obvious: The extra `.catch(done)` at the end of these tests each let the test fail immediately and usefully when an assertion fails, rather than hitting a timeout because `done()` never gets called.